spec/lib/jobs/dell-racadm-tool-job-spec.js

```diff
@@ -0,0 +1,335 @@
+            mockWaterline.nodes.findByIdentifier = function() {
+            };
+            this.sandbox = sinon.sandbox.create();
+            this.sandbox.stub(job, '_subscribeActiveTaskExists').resolves();
+            this.sandbox.stub(mockWaterline.nodes, 'findByIdentifier');
+        });
+
+        afterEach('Dell Racadm Tool Set BIOS Validation', function() {
+            this.sandbox.restore();
+        });
+
+        it('should fail if node does not exist', function() {
+            mockWaterline.nodes.findByIdentifier.resolves(null);
+            return expect(job.run()).to.be.rejectedWith(Errors.AssertionError,
+                'No node for dell racadm set bios');
+        });
+
+        it('should fail if ipmi obmSetting does not exist', function() {
+            var node = {
+                id: 'bc7dab7e8fb7d6abf8e7d6ac',
+                obmSettings: [
+                    {
+                        config: {
+                        }
+                    }
+                ]
+            };
+            mockWaterline.nodes.findByIdentifier.resolves(node);
+            return expect(job.run()).to.be.rejectedWith(Errors.AssertionError,
+                'No ipmi obmSetting for dell racadm set bios');
+        });
+    });
+
+    describe("Obm setting decrypt", function() {
+        beforeEach(function() {
+            encryption.start();
+            this.sandbox = sinon.sandbox.create();
+        });
+
+        afterEach(function() {
+            encryption.stop();
+            this.sandbox.restore();
+        });
+
+        it("should decrypt a password", function() {
+            var iv = 'vNtIgxV4kh0XHSa9ZJxkSg==';
+            var password = encryption.encrypt('password', iv);
+            var data = {
+                password: password
+            };
+            expect(job.revealSecrets(data)).to.deep.equal({
+                password: 'password'
+            });
+        });
+
+        it("should convert a host mac address to an IP", function() {
+            var macToIP = this.sandbox.stub(lookup, 'macAddressToIp');
+            macToIP.resolves(
+                '10.1.1.2'
+            );
+            var data = {
+                host: '7a:c0:7a:c0:be:ef'
+            };
+            return job.lookupHost(data).should.become({ host: '10.1.1.2'});
+        });
+    });
+
+    describe('DellRacadm Tool Set Bios', function() {
+        var options;
+
+        beforeEach('Dell Racadm Tool Set BIOS Validation', function() {
+            options = {
+                server_username: "onrack",
+                server_password: "onrack",
+                server_filePath: "//1.2.3.4/src/bios.xml",
+                action: "SetBIOS"
+            };
+            job = new racadmToolJob(options, {}, uuid.v4());
+            mockWaterline.nodes.findByIdentifier = function(){};
+
+            this.sandbox = sinon.sandbox.create();
+            this.sandbox.stub(job, '_subscribeActiveTaskExists').resolves();
```

Comment:
Since this stub is need by every test cases, so you can move to the outmost `beforeEach`, don't need to write this in every test suite.